packages/angular_devkit/build_angular/src/utils/index-file/augment-index-html.ts

```diff
@@ -222,20 +222,40 @@ function isString(value: unknown): value is string {
   return typeof value === 'string';
 }
 
-async function getLanguageDirection(lang: string, warnings: string[]): Promise<string | undefined> {
+async function getLanguageDirection(
+  locale: string,
+  warnings: string[],
+): Promise<string | undefined> {
+  const dir = await getLanguageDirectionFromLocales(locale);
+
+  if (!dir) {
+    warnings.push(
+      `Locale data for '${locale}' cannot be found. 'dir' attribute will not be set for this locale.`,
+    );
+  }
+
+  return dir;
+}
+
+async function getLanguageDirectionFromLocales(locale: string): Promise<string | undefined> {
   try {
     const localeData = (
       await loadEsmModule<typeof import('@angular/common/locales/en')>(
-        `@angular/common/locales/${lang}`,
+        `@angular/common/locales/${locale}`,
       )
     ).default;
 
     const dir = localeData[localeData.length - 2];
 
     return isString(dir) ? dir : undefined;
   } catch {
-    warnings.push(
-      `Locale data for '${lang}' cannot be found. 'dir' attribute will not be set for this locale.`,
-    );
+    // In some cases certain locales might map to files which are named only with language id.
+    // Example: `en-US` -> `en`.
+    const [languageId] = locale.split('-', 1);
```

Comment:
Will the dir attribute always be on the primary subtag? I always get confused with the localization standard, so I'm sure what applies here, but I think it is possible for [multiple dashes to be included](https://en.wikipedia.org/wiki/IETF_language_tag#:~:text=Up%20to%20three%20optional%20extended%20language%20subtags%20composed%20of%20three%20letters%20each%2C%20separated%20by%20hyphens%3B):

> Up to three optional extended language subtags composed of three letters each, separated by hyphens

Is there a possible case where the locale is something like `zh-Hans-SG`, and the dir attribute is set on `zh-Hans` rather than `zh`?

Reply:
The first item is actual the language Id, which is always the first.

We are actually using this methodology in other places https://github.com/angular/angular-cli/blob/033c4415efd6bb4a073018da446643c1abbe3a40/packages/angular_devkit/build_angular/src/utils/i18n-options.ts#L198